packages/lesswrong/themes/themeNames.ts

```diff
@@ -52,25 +71,51 @@ export function isValidSerializedThemeOptions(options: string|object): boolean {
   }
 }
 
-export function isValidThemeName(name: string): name is UserThemeName {
-  for (let theme of themeMetadata) {
-    if (theme.name === name)
-      return true;
-  }
-  return false;
-}
+export const isValidUserThemeSetting = (name: string): name is UserThemeSetting =>
+  userThemeSettings.includes(name as unknown as UserThemeSetting);
+
+export const resolveThemeName = (
+  theme: UserThemeSetting,
+  prefersDarkMode: boolean,
+): UserThemeName =>  theme === "auto"
+  ? (prefersDarkMode ? "dark" : "default")
+  : theme;
+
+export const abstractThemeToConcrete = (
+  theme: AbstractThemeOptions,
+  prefersDarkMode: boolean,
+): ThemeOptions => themeOptionsAreConcrete(theme)
+  ? theme
+  : {...theme, name: prefersDarkMode ? "dark" : "default"};
 
-export function getForumType(themeOptions: ThemeOptions) {
+export function getForumType(themeOptions: AbstractThemeOptions) {
   const actualForumType = forumTypeSetting.get();
   return (themeOptions?.siteThemeOverride && themeOptions.siteThemeOverride[actualForumType]) || actualForumType;
 }
 
-export const defaultThemeOptions = {"name":"default"};
+export const defaultThemeOptions = {name: forumTypeSetting.get() === "EAForum" ? "auto" : "default"};
+
+const deserializeThemeOptions = (themeOptions: object | string): AbstractThemeOptions => {
+  if (typeof themeOptions === "string") {
+    return isValidUserThemeSetting(themeOptions)
+      ? {name: themeOptions}
+      : JSON.parse(themeOptions);
+  } else {
+    return themeOptions as AbstractThemeOptions;
+  }
+}
+
+export function getThemeOptions(themeCookie: string | object, user: DbUser|UsersCurrent|null): AbstractThemeOptions {
+  // Check if the user setting is a serialized ThemeOptions object
+  if (user?.theme && isValidSerializedThemeOptions(user.theme)) {
+    return deserializeThemeOptions(user.theme);
+  }
```

Comment:
I believe there's an issue here where once a LW dev sets their theme while testing the ea forum, they will never be able to change their settings again on localhost.

Reply:
Also, watch out, did LW use to store it in the user object? We're inverting the order of the check.

Reply:
Good point - now we're deleting the token on the EAForum, it's safe to switch the order back for both.